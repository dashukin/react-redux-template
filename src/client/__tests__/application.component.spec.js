import React from 'react';
import { shallow } from 'enzyme';
import Application from 'src/client/application.component';
import { NOT_FOUND } from 'redux-first-router';
import HomePage from 'src/client/pages/home';
import NotFoundPage from 'src/client/pages/not-found';

const homePage = 'HomePage';
const notFoundPage = 'NotFoundPage';

describe('application.component', () => {
  it('should render Home page as a default one', () => {
    const wrapper = shallow(<Application />);

    expect(wrapper.find(HomePage).length).toBe(1);
  });

  it('should render Home page WHEN appropriate page property is passed', () => {
    const wrapper = shallow(<Application />);

    expect(wrapper.find(HomePage).length).toBe(1);
  });

  it('should render page not found WHEN when appropriate page property is passed', () => {
    const wrapper = shallow(<Application page={notFoundPage} />);

    expect(wrapper.find(NotFoundPage).length).toBe(1);
  });

  describe('application.component snapshots', () => {
    it('should match snapshot with Home page as default one', () => {
      const wrapper = shallow(<Application />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with Home page', () => {
      const wrapper = shallow(<Application page={homePage} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with Not Found page', () => {
      const wrapper = shallow(<Application page={NOT_FOUND} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
