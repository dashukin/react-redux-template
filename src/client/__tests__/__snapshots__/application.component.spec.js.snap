// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application.component application.component snapshots should match snapshot with Home page 1`] = `
<div
  className="application"
>
  <UniversalComponent />
</div>
`;

exports[`application.component application.component snapshots should match snapshot with Home page as default one 1`] = `
<div
  className="application"
>
  <UniversalComponent />
</div>
`;

exports[`application.component application.component snapshots should match snapshot with Not Found page 1`] = `
<div
  className="application"
>
  <UniversalComponent />
</div>
`;
